stl/inc/stdatomic.h

```diff
@@ -7,17 +7,19 @@
 #ifndef _STDATOMIC_H_
 #define _STDATOMIC_H_
 
-#if defined(RC_INVOKED) || defined(Q_MOC_RUN) || defined(__midl)
-// do nothing, see _STL_COMPILER_PREPROCESSOR in yvals_core.h
-#else // ^^^ non-compiler tools / C and C++ compilers vvv
+// do nothing for preprocessors, see _STL_COMPILER_PREPROCESSOR in yvals_core.h
+#if !defined(RC_INVOKED) && !defined(Q_MOC_RUN) && !defined(__midl)
+
 // provide a specific error message for C compilers, before the general error message in yvals_core.h
 #ifndef __cplusplus
 #error <stdatomic.h> is not yet supported when compiling as C, but this is planned for a future release.
 #endif // __cplusplus
-#endif // ^^^ C and C++ compilers ^^^
 
 #include <yvals.h>
-#if _STL_COMPILER_PREPROCESSOR
+// because this header needs to work in C and C++ we do the
+// preprocessor check above, if it's less restrictive than
+// what yvals_core would do then that's a bug.
+_STL_INTERNAL_STATIC_ASSERT(_STL_COMPILER_PREPROCESSOR);

```

Comment:
This comment is slightly confusing (as the header doesn't work in C, it intentionally emits an error). Additionally, `_STL_INTERNAL_STATIC_ASSERT` is active for our test suite only, so this isn't really verifying anything. I recommend simply removing this comment and internal static assert.